src/hooks/BookAppointment/useAppointmentCalendar.hook.tsx

```diff
@@ -1,37 +1,181 @@
+  const formattedThreeMothPeriod = ThreeMothPeriod.map((date) => {
+    date.setHours(0, 0, 0, 0);
+    return date.getTime();
+  });
+
+  const formattedFreeSlots = freeSlots.map((item) => {
+    const date = new Date(item.start);
+    date.setHours(0, 0, 0, 0);
+    return date.getTime();
+  });
+
+  const allBookedDates = formattedThreeMothPeriod.filter(
+    (date) => !formattedFreeSlots.includes(date)
+  );
 
-  const [showCalendar, setShowCalendar] = useState(false);
+  //array with not avalible dates
+  const formattedBookedDates = allBookedDates.map((date) => {
+    const d = new Date(date);
+    const year = d.getFullYear();
+    const month = d.getMonth();
+    const day = d.getDate();
+    return new Date(year, month, day);
+  });
 
-  // Todo :  hardcode, just an example till connection with backend
   const isDisabled = (day: Date) => {
     return day > new Date();
   };
 
   const bookedDays = [
-    new Date(2023, 4, 10),
-    new Date(2023, 4, 12),
-    new Date(2023, 5, 20),
+    ...formattedBookedDates,
     { before: new Date(), modifiers: { disabled: isDisabled } },
   ];
 
-  // Todo: styles for days without empty slots. not sure if we need them at this moment, maybe in next sprint will be deleted or moved to styles
-  const bookedStyle = { color: '#808080' };
+  const selectedDate = new Date(selectedDay);
+
+  const filteredSlots = freeSlots.filter((slot) => {
+    const slotDate = new Date(slot.start);
+    return (
+      slotDate.getDate() === selectedDate.getDate() &&
+      slotDate.getMonth() === selectedDate.getMonth() &&
+      slotDate.getFullYear() === selectedDate.getFullYear()
+    );
+  });
+
+  // Sort all slots by time and leave only unique slots
+  filteredSlots.sort((a, b) => {
+    const timeA = new Date(a.start).getTime();
+    const timeB = new Date(b.start).getTime();
+    return timeA - timeB;
+  });
+
+  const uniqueSlots = [];
+  const uniqueKeys = new Set();
+
+  for (const slot of filteredSlots) {
+    const slotDate = new Date(slot.start);
+
+    if (
+      slotDate.getDate() === selectedDate.getDate() &&
+      slotDate.getMonth() === selectedDate.getMonth() &&
+      slotDate.getFullYear() === selectedDate.getFullYear()
+    ) {
+      const key = `${slot.start}-${slot.end}`;
+
+      if (!uniqueKeys.has(key)) {
+        uniqueSlots.push({
+          doctor: slot.doctor.id,
+          start: slot.start,
+          end: slot.end,
+        });
+        uniqueKeys.add(key);
+      }
+    }
+  }
+
+  const transformedSlots = Object.entries(uniqueSlots).map(([key, value]) => ({
+    doctor: value.doctor.id,
+    start: value.start,
+    end: value.end,
+  }));
+
+  function formatTimeRange(startTime, endTime) {
+    const start = new Date(startTime).toLocaleTimeString([], {
+      hour: '2-digit',
+      minute: '2-digit',
+    });
+    const end = new Date(endTime).toLocaleTimeString([], {
+      hour: '2-digit',
+      minute: '2-digit',
```

Comment:
why its string? why its not in constant? you use it 4 time

Reply:
fix